egs/wsj/s5/steps/nnet3/lstm/train.sh

```diff
@@ -643,6 +648,7 @@ while [ $x -lt $num_iters ]; do
   fi
   x=$[$x+1]
   num_archives_processed=$[$num_archives_processed+$this_num_jobs]
+  rm $dir/cache.$x
```

Comment:
this is a bug too, that you are removing $dir/cache.$x after incrementing x.  It means you delete the one you are going to need.
Did you ever test this script?